src/merkle_proof.rs

```diff
@@ -49,60 +51,68 @@ impl MerkleProof {
         &self.merkle_path
     }
 
-    pub fn compile(self, mut leaves: Vec<(H256, H256)>) -> Result<CompiledMerkleProof> {
-        if leaves.is_empty() {
+    pub fn compile(self, leaves: Vec<(H256, H256)>) -> Result<CompiledMerkleProof> {
+        let mut leaves_ord: Vec<(H256Ord, H256)> = leaves
+            .iter()
+            .take(leaves.len())
+            .map(|(k, v)| (H256Ord { inner: k.clone() }, v.clone()))
+            .collect::<Vec<_>>();
+        if leaves_ord.is_empty() {
             return Err(Error::EmptyKeys);
-        } else if leaves.len() != self.leaves_count() {
+        } else if leaves_ord.len() != self.leaves_count() {
             return Err(Error::IncorrectNumberOfLeaves {
                 expected: self.leaves_count(),
-                actual: leaves.len(),
+                actual: leaves_ord.len(),
             });
         }
         // sort leaves
-        leaves.sort_unstable_by_key(|(k, _v)| *k);
+        leaves_ord.sort_unstable_by_key(|(k, _v)| k.clone());
```

Comment:
Although H256 in numext is different than current H256, why not just modify here to save changes?
for example, reverse the key bits here only.